### DISCUSSION TITLE
docs: Using Multiple Schemas with raw query

### DISCUSSION THREAD

**Discussion by gquittet** [mikro-orm/mikro-orm#4390] [General]
With the `mysql2` library, you can switch schema using the `changeUser` function.

I've read the `Using Multiple Schemas` page (https://mikro-orm.io/docs/multiple-schemas) and I haven't found anything about it.

Do you have an idea how can I do that?

When we have a solution, we can add it to the documentation.

**Comment by B4nan** [comment]
There is no support for switching current schema, not sure how that should work? I mean why would you want that and how would you want to use it? I don't have much experience with those things in MySQL, I guess most of the people who need multiple schemas will use postgres instead. Note that we can't really depend on mysql2 APIs, this would have to go through knex first probably.

The "ORM way" is to have one base/default schema you set in the ORM config, if you want to change that, you should have a new ORM instance, as this affects the metadata as well as the driver connection. Or as in the linked guide, you can have statically defined schema in the entity name (and dynamically query them, there is `FindOptions.schema` for that).

**Comment by gquittet** [comment]
Your library is insane! ❤️ 

To resolve simple cases, I can use the Entity Manager (find, findOne, ...)
and to resolve complex cases like the one I had above, I can the knex instance provided by MikroORM.

To do that, I've made a utility function

```typescript
export const getKnex = (): [MySqlConnection, Knex] => [orm.em.getConnection(), orm.em.getKnex()];
```

and I can easily use knex to query multiple schema.

So for people, that are looking for a solution => just use knex provided by MikroORM

I can add example if you want.

For me, we can close the discussion 🙂 

Have a nice day ! ☀️
